pygeoapi/provider/mapscript_.py

```diff
@@ -74,13 +74,22 @@ def __init__(self, provider_def):
 
             file_extension = self.data.split('.')[-1]
 
-            if file_extension in ['shp', 'tif']:
-                LOGGER.debug('Setting built-in MapServer driver')
-                self._layer.data = self.data
+            tileindex = False
```

Comment:
Suggest using `pygeoapi.util.str2bool`:

```python
            tileindex = False
            if 'tileindex' in self.options:
                if str2bool(self.options.get('tileindex', False))
                    self._layer.tileindex = self.data 
```

Reply:
It is fine to use str2bool, but if we don't combine the two conditions, collections that have `tileindex=False` will never enter the `else` code, and therefore won't have a layer assigned. I propose this:

https://github.com/doublebyte1/pygeoapi/blob/17323a9e98633149d207b10deaa5870b8830399e/pygeoapi/provider/mapscript_.py#L77-80